File: NSToukJ/src/GentleClass.java
public class GentleClass implements RespectedClass.BeingRespectful {

    public String greetings(RespectedClass.DayTime daytime, String name) {
        StringBuilder greetings = new StringBuilder();
        switch (daytime) {
            case kMorning:
                greetings.append("Good morning!");
                break;
            case kEvening:
                greetings.append("Good evening!");
                break;
        }
        greetings.append(" Nice to see you, " + name);
        return greetings.toString();
    }

    @Override
    public String contactInfo() {
        return null;
    }


}

File: NSTouk/NSTouk/GentleClass.m
#import "GentleClass.h"

@implementation GentleClass

- (NSString*)greetingsForDayTime:(kDayTime)dayTime name:(NSString*)name {
    NSMutableString* greetings = [NSMutableString string];
    switch (dayTime) {
        case kMorning:
            [greetings appendString:@"Good morning!"];
            break;
        case kEvening:
            [greetings appendString:@"Good evening!"];
            break;
    }
    [greetings appendString:[@" Nice to see you, " stringByAppendingString:name]];
    return [greetings copy];
}

@end